cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/analysis/fsm/State.kt

```diff
@@ -32,64 +32,113 @@ package de.fraunhofer.aisec.cpg.analysis.fsm
  * - [isAcceptingState] indicates if this State accepts the FSM (in our case, this means that the
  * order of statements was correct).
  */
-class State(val name: String, val isStart: Boolean = false, var isAcceptingState: Boolean = false) :
-    Cloneable {
+sealed class State(name: Int, isStart: Boolean = false, isAcceptingState: Boolean = false) {
```

Comment:
Do we really want to restrict a user to an integer as a name? A String would be more readable...